qutebrowser/components/utils/blockutils.py

```diff
@@ -0,0 +1,170 @@
+# vim: ft=python fileencoding=utf-8 sts=4 sw=4 et:
+
+# Copyright 2020 Florian Bruhin (The Compiler) <mail@qutebrowser.org>
+#
+# This file is part of qutebrowser.
+#
+# qutebrowser is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# qutebrowser is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with qutebrowser.  If not, see <http://www.gnu.org/licenses/>.
+
+
+"""Code that is shared between the host blocker and Brave ad blocker."""
+
+import typing
+import os
+import functools
+import threading
+
+from PyQt5.QtCore import QUrl
+
+from qutebrowser.api import downloads, message, config
+
+
+class FakeDownload(downloads.TempDownload):
+
+    """A download stub to use on_download_finished with local files."""
+
+    def __init__(self, fileobj: typing.IO[bytes]) -> None:
+        # pylint: disable=super-init-not-called
+        self.fileobj = fileobj
+        self.successful = True
+
+
+class BlocklistDownload:
```

Comment:
```suggestion
class BlocklistDownloads:
```

to make it clear this doesn't only represent a single download.